README.md
# React Awesome Selector

[![npm version](https://badge.fury.io/js/react-awesome-selector.svg)](https://badge.fury.io/js/react-awesome-selector)
[![Conventional Commits](https://img.shields.io/badge/Conventional%20Commits-1.0.0-yellow.svg)](https://conventionalcommits.org)

A selector component built with ReactJS.

## [Examples (Storybook)](https://statebait.github.io/react-awesome-selector)

## Getting started

To install:

```bash
npm install react-awesome-selector
```

-Or with yarn-

```bash
yarn add react-awesome-selector
```

### Usage

```jsx
import React from 'react'
import Selector from 'react-awesome-selector'
import 'react-awesome-selector/dist/style.css'

const data = [
  { name: 'card', value: 89519 },
  { name: 'array', value: 49024 },
  { name: 'grocery', value: 90170 },
  { name: 'input', value: 56963 },
]

const withCategoryData = [
  { category: 'calculate', name: 'card', value: 89519 },
  { category: 'calculate', name: 'array', value: 49024 },
  { category: 'lavender', name: 'grocery', value: 90170 },
  { category: 'lavender', name: 'input', value: 56963 },
]

export default function Main() {
  return (
    <div>
      <Selector
        data={data}
        selectTitle="Items"
        selectedTitle="Cart"
        onChange={(values) => console.log(values)}
      />
      <Selector
        data={withCategoryData}
        categorize
        selectTitle="Items"
        selectedTitle="Cart"
        onChange={(values) => console.log(values)}
      />
    </div>
  )
}
```

### Props:

```js
Selector.propTypes = {
  /**
   * Data input for the selector
   */
  data: PropTypes.array,
  /**
   * Whether to display categories in the select list
   */
  categorize: PropTypes.bool,
  /**
   * Title for the select list
   */
  selectTitle: PropTypes.string,
  /**
   * Title for the selected list
   */
  selectedTitle: PropTypes.string,
  /**
   * Function for getting the selected values when anything changes
   */
  onChange: PropTypes.func,
}

Selector.defaultProps = {
  data: [],
  categorize: false,
  selectTitle: '',
  selectedTitle: '',
  onChange: () => {},
}
```

### Styling

Use the default styles by importing:

```js
import 'react-awesome-selector/dist/style.css'
```

These classes are available for use:

```css
// Main wrapper

.react-awesome-selector-wrapper {
}

// Category

.react-awesome-selector-category {
}

.react-awesome-selector-category-title {
}

.react-awesome-selector-category-children {
}

// Selected & Select list

.react-awesome-selector-selected-list {
}

.react-awesome-selector-select-list {
}

.react-awesome-selector-select-list {
}

.react-awesome-selector-selected-list-title {
}

.react-awesome-selector-select-list-title {
}

.react-awesome-selector-selected-list-items {
}

// Item

.react-awesome-selector-item {
}

.react-awesome-selector-item-icon {
}
```

## License

This project is licensed under the MIT License - Copyright (c) 2020 Mohamed Shadab


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [2.0.0](https://github.com/statebait/react-awesome-selector/compare/v1.1.0...v2.0.0) (2021-04-19)

- upgraded all deps ([9bb6c5c](https://github.com/statebait/react-awesome-selector/commit/9bb6c5c4d4fef12ebf0cf1bac98076a6db74c1f4))

## [1.1.0](https://github.com/statebait/react-awesome-selector/compare/v1.0.1...v1.1.0) (2020-12-01)

- upgraded all deps including webpack 5 ([9e21451](https://github.com/statebait/react-awesome-selector/commit/9e21451abf2c5a91fca8210ecf3b6065fdb29a9e))

### [1.0.1](https://github.com/statebait/react-awesome-selector/compare/v1.0.0...v1.0.1) (2020-09-12)

### Bug Fixes

- updated deps to fix security issue ([e8a0717](https://github.com/statebait/react-awesome-selector/commit/e8a07171cc709bdae7fd61b84122a5b86004dd9a))

### Documentation

- fixed changelog compare versions ([b9d6c62](https://github.com/statebait/react-awesome-selector/commit/b9d6c620da25e0d69ad7a0df78b68e9126f38924))

## [1.0.0](https://github.com/statebait/react-awesome-selector/compare/v0.3.1...v1.0.0) (2020-08-19)

### ⚠ BREAKING CHANGES

- replace getSelected prop with onChange

### Features

- categories is now optional ([326c9c8](https://github.com/statebait/react-awesome-selector/commit/326c9c87fad366f7de0b5e2621dd28377ff64897))
- replace getSelected prop with onChange ([bfa4b66](https://github.com/statebait/react-awesome-selector/commit/bfa4b66976381637707f70124ef4ccbca857e852))

### Documentation

- update README ([9f8139a](https://github.com/statebait/react-awesome-selector/commit/9f8139a089d64794a45abf33e73a9c1bb243a151))
- updated readme with new props ([e2d67f7](https://github.com/statebait/react-awesome-selector/commit/e2d67f7326710fcff5f2a0dc2eb3617c21791dc8))

### [0.3.1](https://github.com/statebait/react-awesome-selector/compare/v0.3.0...v0.3.1) (2020-08-14)

### Refactor

- cleanup styles file ([04d7c7e](https://github.com/statebait/react-awesome-selector/commit/04d7c7e4f35d9a553c1346bbcc4a26ee58a8b538))
- useContext hook usage ([0c88501](https://github.com/statebait/react-awesome-selector/commit/0c88501d9897bc9960d122e583ef37dfe869c373))

## [0.3.0](https://github.com/statebait/react-awesome-selector/compare/v0.2.1...v0.3.0) (2020-08-14)

### Performance

- refactor data flow for better perf ([cbf6adf](https://github.com/statebait/react-awesome-selector/commit/cbf6adf800373b2a1220f45e3467ba2fdb4835ff))

### [0.2.1](https://github.com/statebait/react-awesome-selector/compare/v0.2.0...v0.2.1) (2020-08-13)

### Refactor

- additional cleanup ([57cfaf5](https://github.com/statebait/react-awesome-selector/commit/57cfaf5b35a7f8e854160d84bf076cb56bafdfd3))

## [0.2.0](https://github.com/statebait/react-awesome-selector/compare/v0.1...v0.2.0) (2020-08-12)

### Refactor

- rewrote all components with hooks ([d2289f3](https://github.com/statebait/react-awesome-selector/commit/d2289f3843bfa2a86230bdbd0ddb402916dc44e3))

### Bug Fixes

- updated deps to fix security issues ([a26daca](https://github.com/statebait/react-awesome-selector/commit/a26dacaa3bc2698fd10cfb9c52d19dca7ba22616))

### [0.1.5](https://github.com/statebait/react-awesome-selector/compare/v0.1.4...v0.1.5) (2020-06-19)

Stub release

### [0.1.4](https://github.com/shadxx7/react-awesome-selector/compare/v0.1.3...v0.1.4) (2020-04-29)

### Bug Fixes

- **deps:** updated deps for security ([5b312c2](https://github.com/shadxx7/react-awesome-selector/commit/5b312c26b87edd8aea43130b3dc2ee050e7cb2f2))

### [0.1.3](https://github.com/shadxx7/react-awesome-selector/compare/v0.1.2...v0.1.3) (2019-07-16)

### Bug Fixes

- 🐛 updated lodash to fix vulnerabilities ([636f76b](https://github.com/shadxx7/react-awesome-selector/commit/636f76b))

### [0.1.2](https://github.com/shadxx7/react-awesome-selector/compare/v0.1.1...v0.1.2) (2019-07-11)

### Bug Fixes

- 🐛 updated dependencies to fix vulnerabilities ([01259a0](https://github.com/shadxx7/react-awesome-selector/commit/01259a0))

### [0.1.1](https://github.com/shadxx7/react-awesome-selector/compare/v0.1.0...v0.1.1) (2019-06-25)

### Bug Fixes

- 🐛 importing component fix ([785e4fc](https://github.com/shadxx7/react-awesome-selector/commit/785e4fc))

## 0.1.0 (2019-06-10)

- Initial release


stories/index.stories.js
import React from 'react'
import Selector from '../src'
import './utils/wdyr'

export default {
  title: 'Selector',
  component: Selector,
}

const data = [
  { category: 'Calculate', name: 'Card', value: 89519 },
  { category: 'Calculate', name: 'Array', value: 49024 },
  { category: 'Calculate', name: 'Revolutionary', value: 80686 },
  { category: 'Calculate', name: 'Movies', value: 54947 },
  { category: 'Calculate', name: 'Innovative', value: 50595 },
  { category: 'Lavender', name: 'Grocery', value: 90170 },
  { category: 'Lavender', name: 'Input', value: 56963 },
  { category: 'Lavender', name: 'Ergonomic Metal Mouse', value: 46594 },
  { category: 'Lavender', name: 'Dynamic', value: 98821 },
  { category: 'Lavender', name: 'Consultant', value: 48437 },
  { category: 'Home Loan Account', name: 'Roads', value: 67238 },
  { category: 'Home Loan Account', name: 'Corners', value: 42862 },
  { category: 'Home Loan Account', name: 'Rubber', value: 28415 },
  { category: 'Home Loan Account', name: 'HTTP', value: 74444 },
]

export const WithDataOnly = () => (
  <Selector data={data} onChange={(values) => console.log(values)} />
)

export const WithCustomTitle = () => (
  <Selector
    data={data}
    selectTitle="Items"
    selectedTitle="Cart"
    onChange={(values) => console.log(values)}
  />
)

export const WithCategories = () => {
  return (
    <Selector
      data={data}
      categorize
      selectTitle="Items"
      selectedTitle="Cart"
      onChange={(values) => console.log(values)}
    />
  )
}
